.ci/containers/membership-checker/reviewer_assignment.go

```diff
@@ -0,0 +1,206 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+)
+
+func reviewerAssignment(author, prNumber, GITHUB_TOKEN string) error {
+	if isNoAssigneeUser(author) {
+		fmt.Println("User is on the list, not assigning")
+		return nil
+	}
+
+	requestedReviewer, err := getPullRequestRequestedReviewer(prNumber, GITHUB_TOKEN)
+	if err != nil {
+		return err
+	}
+
+	previousAssignedReviewers, err := getPullRequestPreviousAssignedReviewers(prNumber, GITHUB_TOKEN)
+	if err != nil {
+		return err
+	}
+
+	if requestedReviewer != "" {
+		fmt.Println("Issue is assigned")
+		if previousAssignedReviewers != nil {
+			fmt.Println("Retrieving previous reviewers to re-request reviews")
+			for _, reviewer := range previousAssignedReviewers {
+				if isTeamReviewer(reviewer) {
+					err = assignPullRequestReviewer(prNumber, reviewer, GITHUB_TOKEN)
+					if err != nil {
+						return err
+					}
+				}
+			}
+		}
+		return nil
+	}
+
+	if previousAssignedReviewers == nil {
```

Comment:
This section could be simplified by doing something like

```go
foundTeamReviewer := false
if hasTeamReviewer || previousAssignedReviewers != nil {
    if hasTeamReviewer {
         foundTeamReviewer = true
    }
    // check previous reviewers for team reviewers & re-request review from them & set foundTeamReviewer appropriately
}
if !foundTeamReviewer {
    // choose random reviewer
}
```

`hasTeamReviewer` would fill a role similar to getPullRequestRequestedReviewer with one critical difference: it needs to check whether any of the people who have left reviews are team reviewers - that is, if an off-team person left a review, that can't fulfill the requirement of "at least one reviewer from the team".